edx_proctoring/api.py

```diff
@@ -690,6 +690,20 @@ def create_exam_attempt(exam_id, user_id, taking_as_proctored=False):
             )
             log.error(log_msg)
 
+    if not external_id:
+        # temporary logging added to better understand cases where we end up
+        # with exam attempts with NULL external_ids - a situation, that, theoretically,
+        # shouldn't be occurring (EDUCATOR-4948)
+        log_msg = (
+            u'Registered exam attempt with an external ID {external_id}'
```

Comment:
I don't feel that this log should be permanent. If you look up a few lines at line 655, you'll see that we're handling an exception `BackendProviderSentNoAttemptID`, which is raised when we don't get back an `external_id` from the provider. This is the what is supposed to handle the error case of a null `external_id`, but, for some unknown reason, that exception handler is not being hit, which is troublesome. For that reason, this log just serves to aid in debugging why that's happening. It could be that there's another code path we haven't considered that results in no `external_id` or it could be something else. In either case, we don't know what the error is, so I don't think this is a permanent error detection. Does that make sense?

Reply:
I elected to just print out the value of the `external_id` just to be absolutely sure of its value. Our API contract with the provider specifies that they must return an `external_id`, so there's no guarantees on what this value could be if it's not being passed through in their response. I want to be sure there's not some strange case where the provider is sending back a value we aren't handling correctly.